src/OrbitGl/TimerTrack.cpp

```diff
@@ -106,6 +107,14 @@ void TimerTrack::DrawTimesliceText(TextRenderer& text_renderer,
       GlCanvas::kZValueBox, formatting, elapsed_time_length);
 }
 
+[[nodiscard]] inline Tetragon TetragonWithZ(const Tetragon& tetragon, float z) {
```

Comment:
Actually, isn't it simpler to include both the tetragon and also the 'z' in AddTetragonBorder method and erase this method?

Reply:
Should I rather call it `GetTetragonWithNewZ` ?
I don't like to pass a Tetragon and a `float z` together anywhere. As it would effectively mean to pass two z-values and this would be a source of confusion.

Reply:
I have an opposite opinion. We are assuming every shape have the same z in the 4 vertices that can be not true, I would rather refactor methods to use Vec2 and add the z as a parameter instead in all the methods. This is a step backwards.

Reply:
Agree, I never liked Vec3 to be honest. Should we at least make Tetragon store four Vec2 instances and `float z`?